src/resolvers/Mutation/updateUserTag.ts

```diff
@@ -0,0 +1,92 @@
+import { MutationResolvers } from "../../types/generatedGraphQLTypes";
+import { errors, requestContext } from "../../libraries";
+import { User, OrganizationTagUser } from "../../models";
+import {
+  USER_NOT_FOUND_ERROR,
+  USER_NOT_AUTHORIZED_ERROR,
+  TAG_NOT_FOUND,
+  NO_CHANGE_IN_TAG_NAME,
+  TAG_ALREADY_EXISTS,
+} from "../../constants";
+
+export const updateUserTag: MutationResolvers["updateUserTag"] = async (
+  _parent,
+  args,
+  context
+) => {
+  const currentUser = await User.findOne({
+    _id: context.userId,
+  }).lean();
+
+  // Checks whether currentUser exists.
+  if (!currentUser) {
+    throw new errors.NotFoundError(
+      requestContext.translate(USER_NOT_FOUND_ERROR.MESSAGE),
+      USER_NOT_FOUND_ERROR.CODE,
+      USER_NOT_FOUND_ERROR.PARAM
+    );
+  }
+
+  // Get the tag object
+  const tag = await OrganizationTagUser.findOne({
+    _id: args.input._id,
+  }).lean();
+
+  if (!tag) {
+    throw new errors.NotFoundError(
+      requestContext.translate(TAG_NOT_FOUND.MESSAGE),
+      TAG_NOT_FOUND.CODE,
+      TAG_NOT_FOUND.PARAM
+    );
+  }
+
+  // Throw error if the new tag name is the same as the old one
+  if (tag.name === args.input.name) {
+    throw new errors.ConflictError(
+      requestContext.translate(NO_CHANGE_IN_TAG_NAME.MESSAGE),
+      NO_CHANGE_IN_TAG_NAME.CODE,
+      NO_CHANGE_IN_TAG_NAME.PARAM
+    );
+  }
+
+  // Check if another tag with the new name exists under the same parent tag
+  const anotherTagExists = await OrganizationTagUser.exists({
+    name: args.input.name,
+    parentTagId: tag!.parentTagId,
+  });
+
+  if (anotherTagExists) {
+    throw new errors.ConflictError(
+      requestContext.translate(TAG_ALREADY_EXISTS.MESSAGE),
+      TAG_ALREADY_EXISTS.CODE,
+      TAG_ALREADY_EXISTS.PARAM
+    );
+  }
+
+  // Boolean to determine whether user is an admin of organization of the tag folder.
+  const currentUserIsOrganizationAdmin = currentUser.adminFor.some(
```

Comment:
This authorization check should happen right after you confirm whether the tag exists.